internal/host/hostutil/host_utils_test.go

```diff
@@ -266,6 +268,154 @@ var _ = DescribeTable("IsDiskEncryptionEnabledForRole", func(enabledOn string, r
 	Entry("enabledOn none, role worker", models.DiskEncryptionEnableOnNone, models.HostRoleWorker, false),
 )
 
+var _ = Describe("GetHostInstallationDisk", func() {
+	var (
+		hostId    strfmt.UUID
+		diskId    = "/dev/disk/by-id/test-disk"
+		diskName  = "test-disk"
+		validHost *models.Host
+	)
+
+	BeforeEach(func() {
+		hostId = strfmt.UUID(uuid.New().String())
+	})
+
+	It("should return installation disk when found by disk ID", func() {
+		inventory := &models.Inventory{
+			Disks: []*models.Disk{
+				{ID: diskId, Name: diskName},
+			},
+		}
+		inventoryBytes, _ := json.Marshal(inventory)
+		validHost = &models.Host{
+			ID:                   &hostId,
+			Inventory:            string(inventoryBytes),
+			InstallationDiskID:   diskId,
+			InstallationDiskPath: "",
+		}
+
+		disk, err := GetHostInstallationDisk(validHost)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(disk).NotTo(BeNil())
+		Expect(disk.ID).To(Equal(diskId))
+		Expect(disk.Name).To(Equal(diskName))
+	})
+
+	It("should return installation disk when found by disk path", func() {
+		// GetDeviceFullName returns "/dev/{diskName}", so we need to use that for the installation path
+		expectedFullName := fmt.Sprintf("/dev/%s", diskName) // This will be "/dev/test-disk"
```

Comment:
nit: the code it's quite self-explanatory, one comment it's acceptable if you want to be explicit, but two seems excessive :)